### DISCUSSION TITLE
Download an OS list items vanished

### DISCUSSION THREAD

**Discussion by frankemb** [cockpit-project/cockpit-machines#1840] [Q&A]
Hello. 

I noticed that several distros available for download vanished from the "Operating system" drop list. 

![image](https://github.com/user-attachments/assets/8224bc97-bb0f-43f6-a757-ca9b4aed8f48)

Now there's none ubuntu versions and for debian only "Debian Testing" is available. 

It happened suddenly, without any update or changes of any kind. 

which could be the cause and how could I update that list?

versions:
         debian 12 bookwork
         cockpit 287
         cockpit-machines 288

**Comment by martinpitt** [comment]
This is mostly a libosinfo thing. It may only offer distros which are not EOL?

**Reply by garrett** [reply]
This is correct; it shows distros that are not EOL with a bit of a grace period. There's a feature to override this and show all of them, but it hasn't been implemented yet — only spec'd out.

- https://github.com/cockpit-project/cockpit-machines/issues/509
- https://github.com/cockpit-project/cockpit-machines/issues/1260

I also summarized 509 @ this comment:
- https://github.com/cockpit-project/cockpit-machines/issues/1181#issuecomment-1934473220

(But, generally, people should avoid using EOL OSes, as they're going to have security problems once they're on a network.)

**Reply by garrett** [reply]
Note, though, that Debian 12 Bookworm is supported until mid-2028, so it should definitely be in the list:

"Debian 12.7 was released on [August 31st, 2024](https://www.debian.org/News/2024/20240831). Debian 12.0 was initially released on June 10th, 2023. [...]

The Debian 12 life cycle encompasses five years: the initial three years of full Debian support, until June 10th, 2026, and two years of Long Term Support (LTS), until June 30th, 2028"

from https://www.debian.org/releases/bookworm/

**Reply by flx5** [reply]
So apparently the EOL date for Debian 12 was added in december of 2023, however the version of libosinfo distributed by debian bookworm seems to contain the source files from february 2022: https://packages.debian.org/bookworm/libosinfo-1.0-0.

The package does not seem to be available in backports either so it seems stable currently is stuck with the old version.

**Reply by flx5** [reply]
One workaround for now is to do place the attached debian-12.xml in /usr/share/osinfo/os/debian.org/debian-12.xml.
(This is the same xml as https://gitlab.com/libosinfo/osinfo-db/-/blob/9ecec7e36da46878535e752f8040169f47975995/data/os/debian.org/debian-12.xml.in) but with added translations by running `make data`.

[debian-12.zip](https://github.com/user-attachments/files/18648757/debian-12.zip)
